app/grandchallenge/algorithms/serializers.py

```diff
@@ -90,3 +94,85 @@ class HyperlinkedJobSerializer(JobSerializer):
 
     class Meta(JobSerializer.Meta):
         pass
+
+
+class JobPostSerializer(JobSerializer):
+    algorithm_title = CharField(write_only=True)
+    inputs = ComponentInterfaceValuePostSerializer(many=True)
+
+    class Meta:
+        model = Job
+        fields = ["pk", "algorithm_title", "inputs", "status"]
+
+    def validate(self, data):
+        alg = Algorithm.objects.get(title=data.pop("algorithm_title"))
+        if not alg.latest_ready_image:
+            raise serializers.ValidationError(
+                "Algorithm image is not ready to be used"
+            )
+
+        data["algorithm_image"] = alg.latest_ready_image
+
+        # validate that no inputs are provided that are not configured for the
+        # algorithm and that all interfaces without defaults are provided
+        algorithm_input_ids = set(a.id for a in alg.inputs.all())
+        input_ids = set(i["interface_id"] for i in data["inputs"])
+
+        # surplus inputs: provided but interfaces not configured for the algorithm
+        surplus = ComponentInterface.objects.filter(
+            id__in=list(input_ids - algorithm_input_ids)
+        )
+        if surplus:
+            titles = ", ".join((ci.title for ci in surplus))
+            raise serializers.ValidationError(
+                f"Provided inputs(s) {titles} are not defined for this algorithm"
+            )
+
+        # missing inputs
+        missing = alg.inputs.filter(
+            id__in=list(algorithm_input_ids - input_ids),
+            default_value__isnull=True,
+        )
+        if missing:
+            titles = ", ".join((ci.title for ci in missing))
+            raise serializers.ValidationError(
+                f"Interface(s) {titles} do not have a default value and should be provided."
+            )
+
+        return data
+
+    def create(self, validated_data):
+        inputs_data = validated_data.pop("inputs")
+        job = Job.objects.create(**validated_data)
+        component_interface_values = []
+        upload_pks = {}
+        algorithm_input_ids = set(
+            a.id for a in job.algorithm_image.algorithm.inputs.all()
+        )
+        input_ids = set(i["interface_id"] for i in inputs_data)
+
+        for input_data in inputs_data:
+            # check for upload_pk in input
+            upload_pk = input_data.pop("upload_pk", None)
+            civ = ComponentInterfaceValue.objects.create(**input_data)
```

Comment:
Do we check that the data are valid?

Reply:
Yes, validation of the ComponentInterfaceValues is done in components/serializers.py